modules/arrow/src/geoarrow/convert-geoarrow-to-binary-geometry.ts

```diff
@@ -99,8 +99,8 @@ export function getBinaryGeometriesFromArrow(
         ...BINARY_GEOMETRY_TEMPLATE,
         ...(featureTypes.polygon ? binaryContent : {}),
         polygonIndices: {
-          // TODO why deck.gl's tessellatePolygon performance is not good when using geometryIndicies
-          // even when there is no hole in any polygon
+          // NOTE: polygonIndices and primitivePolygonIndices are not used together to render polygon (binary) with holes
+          // in deck.gl currently, so we pass geomOffset as a work around. This issue needs to be fixed in deck.gl
```

Comment:
I am confused about this. 
- What is the correct thing to pass? An array with zero elements? 
- What is in the work around? `geomOffsets` can we name this better or comment on it? Why does that work for deck?

Reply:
From the doc (https://deck.gl/docs/api-reference/layers/geojson-layer#polygons), we should pass `polygonIndices` and `primitivePolygonIndices` together to deal with the case that polygon has any holes.  

Correct me if I am wrong: when I read the code in deck.gl, it seems this only happens in carto-vector-tile-loader (see [here](https://github.com/visgl/deck.gl/blob/b5108c8904797732a83eabb1cd3cdca2f71f6ee4/modules/carto/src/layers/schema/carto-vector-tile-loader.ts#L68)), not in solid-polygon-layer with binary geometry format. They are simply used for polygon and polygonOutline indicies (see [here](https://github.com/visgl/deck.gl/blob/master/modules/layers/src/geojson-layer/geojson-layer-props.ts))

Thanks, @ibgreen!

Reply:
I created another PR https://github.com/visgl/loaders.gl/pull/2785 to add triangle indices using earcut, so we can pass `geomOffsets` for both polygonIndices and primitivePolygonIndices along with the `triangles` props for solid-polygon-layer with binary geometry format.

Reply:
> Correct me if I am wrong: when I read the code in deck.gl, it seems this only happens in carto-vector-tile-loader (see [here](https://github.com/visgl/deck.gl/blob/b5108c8904797732a83eabb1cd3cdca2f71f6ee4/modules/carto/src/layers/schema/carto-vector-tile-loader.ts#L68)), 

@felixpalmer I see your name on those lines in carto module. Is that something that could/should be upstreamed to the generic / non-carto module?

Reply:
@lixun910 FWIW - when I ask for clarification, I am usually not asking for additional detail in the PR comments, but for additional detail in the source code comments, so that things are easier for me to understand when I come back to this code in 6 months.

Reply:
The reason we have the triangulation in the carto module is so that we can perform it in our worker. As I commented on the deck.gl repo, `SolidPolygonLayer` [should support](https://deck.gl/docs/api-reference/layers/solid-polygon-layer/#use-binary-attributes) triangulation with binary data automatically.

My experience is that triangulation can be a very costly operation, so if it can be performed in the worker thread in loaders.gl that will likely give the best performance. Once you put it into deck, it is being done on the main thread and will lead to jank